app/serializers/oauth_userinfo_serializer.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class OauthUserinfoSerializer < ActiveModel::Serializer
+  include RoutingHelper
+
+  attributes :iss, :sub, :name, :preferred_username, :profile, :picture
+
+  def iss
+    root_url
+  end
+
+  def sub
+    ActivityPub::TagManager.instance.uri_for(object)
```

Comment:
@ClearlyClaire did the logic of using the URI for the Account make sense here? This can be either a String or a URI, so we could've made it the user's account ID, but in my experience, that's often not unique on hosts like MastoHost who automate the provisioning of the initial user account. (This bit us when building FediCheck)

Reply:
Account IDs are neither globally consistent nor globally unique, so I think using the URI for this is sensible.

Reply:
We may want an accounts API that takes in the account URI and redirects to the account ID representation, just to provide an upgrade path in the future? I don't think the AP Actor provides the Account ID which is necessary for other operations with the mastodon API.

Reply:
So like: `GET /api/v1/accounts/lookup` with a URI parameter

Reply:
Wait, when/why do we want it, and how does it relate to this PR?

Reply:
Okay, so, I initially use `profile` and `/oauth/userinfo`, I then decide I need something more, how can I go from the data I already have to the fuller Account objects?

I can get the Actor by requesting the `sub` as `application/ld+json` (potentially with authorized fetch), but not the Account entity.

Reply:
Is there a reason in that scenario you could not just use `/api/v1/accounts/credentials`? Both would require `profile` and be Mastodon-specific.

Reply:
Yes, so if you've previously used userinfo to get data, but then want to change UI and show more information, you can't backfill it with the data you already have, because the mastodon account ID isn't present in the userinfo response, so you'd need a way to go from Account URI => Account ID to enhance your existing data without requiring the user reauthenticates

Reply:
But querying `/api/v1/accounts/credentials` does not require you to re-authenticate? You can re-use the same token you just used?

Reply:
Right, but only if you've *just* used it, the time between getting userinfo and your application changing, the user may have revoked access or the access token may have expired (assuming a world with token expiration)

Reply:
I should note, this conversation doesn't block this PR, I don't think. There are ways to solve this in the future, if needed